android/src/main/java/com/twiliorn/library/CustomTwilioVideoView.java

```diff
@@ -1362,12 +1385,47 @@ private RemoteDataTrack.Listener remoteDataTrackListener() {
 
             @Override
             public void onMessage(RemoteDataTrack remoteDataTrack, ByteBuffer byteBuffer) {
+                Log.d("HEYYY", "HEYYY ONMESSAGE BYTEBUFFER");
+//                Log.d("HEYYY", String.valueOf(byteBuffer));
+                if (byteBuffer.hasArray()) {    // don't need this
+//                    Log.d("HEYYY", "HAS ARRAY");
+                    String decodedString = new String(byteBuffer.array(), StandardCharsets.UTF_8);
```

Comment:
As I mentioned on Slack, this line of code should be removed. You have it right in the `else` block below. You just need to do the same thing in this `if` block. The code between the if/else blocks should be almost identical. I think if you refactor it'll hopefully make more sense:

```java
byte[] binaryMessage;

if (byteBuffer.hasArray()) {
    binaryMessage = byteBuffer.array();
} else {
    binaryMessage = new byte[byteBuffer.remaining()];
    byteBuffer.get(binaryMessage);
}

String encodedBinaryMessage = Base64.encodeToString(binaryMessage, Base64.NO_WRAP);
WritableMap event = buildDataTrackEvent(remoteDataTrack, encodedBinaryMessage);
pushEvent(CustomTwilioVideoView.this, ON_DATATRACK_BINARY_MESSAGE_RECEIVED, event);
```